FinSearchUnified/Subscriber/Widgets.php

```diff
@@ -33,4 +93,61 @@ public function beforeListingCountAction(Enlight_Hook_HookArgs $args)
             }
         }
     }
+
+    /**
+     * @param Enlight_Controller_Request_RequestHttp $request
+     *
+     * @return string
+     * @throws Exception
+     */
+    private function parseReferer($request)
+    {
+        $referrer = $request->headers->get('referer');
+        $url = parse_url($referrer, PHP_URL_PATH);
+
+        $basePath = $request->getBasePath();
+        if ($basePath !== '') {
+            $basePath = rtrim($basePath, '/') . '/';
+            $url = str_replace($basePath, '', $url);
+        }
+        $url = ltrim($url, '/');
+
+        return $url;
+    }
+
+    /**
+     * Method to match the given URL if it is a category page and set the isCategoryPage session variable accordingly
+     *
+     * @param string $url
+     */
+    public function matchCategoryPageUrl(string $url)
```

Comment:
Consider returning a boolean here. This way the method could be renamed to `isCategoryPage`. The user of this method just sets `Shopware()->Session()->isCategoryPage = $this->isCategoryPage()`. This not only good for semantic purposes but also reduces side effects of this method.